src/util/pelz_io.c

```diff
@@ -247,17 +263,107 @@ int read_pipe(char *msg)
       {
       case 'c':
         len = strcspn(msg, "\n");
-        path = new_charbuf(len - 11); //the number 11 is used because it the number of chars in "pelz -l -c "
-        memcpy(path.chars, &msg[11], (path.len));
-        free_charbuf(&path);
-        pelz_log(LOG_INFO, "Load cert call not added");
+        path = (char *) malloc((len - 10) * sizeof(char));  //the number 10 is used because it the number of chars in "pelz -l -c " minus 1 for the null terminator
+        memcpy(path, &msg[11], len);  //the number 11 is used because it the number of chars in "pelz -l -c "
+
+        if (read_bytes_from_file(path, &data, &data_length))
+        {
+          pelz_log(LOG_ERR, "Unable to read file %s ... exiting", path);
+          free(path);
+          return 1;
+        }
+
+        pelz_log(LOG_DEBUG, "Read bytes from file %s", path);
+        memcpy(path_ext, &path[len - 15], 4); //4 is the set length of .nkl and .ski and 15 is the 11 deducted along with the 4 from the ext
```

Comment:
This memcpy assumes all extensions are four character long. This can lead to misleading/incorrect debug statements, such as:

$ ./bin/pelz load cert README.md

    ...
    pelz-0.0.0 DEBUG - src/util/pelz_io.c(read_pipe:278) Path_ext: E.md
    ...

Reply:
In fact, going the other direction:

    $ ./bin/pelz load cert example.txtxtxtxtxtxt


causes a seg fault. I am inclined to think that some of the file checking should be done by pelz instead of pelz-service.